qiskit_ibm/runtime/utils.py

```diff
@@ -158,12 +231,21 @@ def default(self, obj: Any) -> Any:  # pylint: disable=arguments-differ
             return {'__type__': 'spmatrix', '__value__': value}
         return super().default(obj)
 
+    def encode(self, obj: Any) -> str:  # pylint: disable=arguments-differ
+        if isinstance(obj, (dict, list)):
+            obj = _set_int_keys_flag(obj)
+        encoded = super().encode(obj)
+        if isinstance(obj, (dict, list)):
+            obj = _remove_int_keys_flag(obj)
```

Comment:
For the generic case I am not really sure what the time complexity of either is, but deepcopy is not cheap. I will use deepcopy just for the settings scenario for this ticket since I feel settings won't be huge.